### PR TITLE
chore: add python notebooks for remaining endpoints

### PR DESCRIPTION
**Pull request by danieljhegeman**
Also includes a few small improvements/mods to previously-added notebooks as well as some minor refactoring of existing code.

### CODE REVIEW

File: notebooks/curation_api/python/delete_dataset.ipynb [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+   "source": [
+    "set_api_access_config(api_key_file_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6085d322",
+   "metadata": {},
+   "source": [
+    "### Create new private Collection"
```
Comment (nayib-jose-gloria) [inline notebooks/curation_api/python/delete_dataset.ipynb:131]: Rename to Delete Dataset

File: notebooks/curation_api/python/update_collection.ipynb [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+   "source": [
+    "set_api_access_config(api_key_file_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6085d322",
+   "metadata": {},
+   "source": [
+    "### Fetch Collection"
```
Comment (nayib-jose-gloria) [inline notebooks/curation_api/python/update_collection.ipynb:146]: Rename to Update Collection

File: notebooks/curation_api/python/update_curator_tag.ipynb [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+   "source": [
+    "set_api_access_config(api_key_file_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6085d322",
+   "metadata": {},
+   "source": [
+    "### Create new private Collection"
```
Comment (nayib-jose-gloria) [inline notebooks/curation_api/python/update_curator_tag.ipynb:152]: Rename to Update Curator Tag

File: notebooks/curation_api/python/src/dataset.py [hunk 4]
```diff
@@ -1,18 +1,103 @@
...
+            if dataset_uuid := matches.get("dataset_uuid"):
+                identifier_type = "dataset_uuid"
+                identifier_value = dataset_uuid
+            else:
+                identifier_type = "curator_tag"
+
+    if not identifier_type:
+        raise Exception(f"The identifier '{identifier}' must either 1) include a '.h5ad' suffix OR 2) be a uuid")
+
+    return identifier_value, identifier_type
```
Comment (nayib-jose-gloria) [inline notebooks/curation_api/python/src/dataset.py:46]: super nit: I would prefer returning (type, value) instead to match the function name order

File: notebooks/curation_api/python/src/dataset.py [hunk 5]
```diff
@@ -1,18 +1,103 @@
...
 
 logger = get_custom_logger()
 
+UUID_REGEX = r"[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}"
+DATASET_ID_REGEX = f"(?P<dataset_uuid>{UUID_REGEX})"
+CURATOR_TAG_PREFIX_REGEX = r"(?P<tag_prefix>.*)"
+EXTENSION_REGEX = r"(?P<extension>h5ad)"
+
+
+def get_identifier_type_and_value(identifier: str) -> Tuple[str, str]:
```
Comment (nayib-jose-gloria) [inline notebooks/curation_api/python/src/dataset.py:26]: What do you think of adding a parameter to both delete and patch dataset asking the user to input whether the current identifier they're providing is a curator_tag or dataset_uuid? Presumably, they know the difference and can save us from doing this work on the backend

Comment (nayib-jose-gloria) [reply notebooks/curation_api/python/src/dataset.py:26]: we can add a quick check that their input matches either of the available options, else raise an exception asking for a proper input

**Review by nayib-jose-gloria — CHANGES_REQUESTED** [review]
Requested a couple of doc heading changes + a comment considering a refactor for the newly added dataset functions

File: notebooks/curation_api/python/src/dataset.py (hunk 5 above)
Comment (danieljhegeman) [reply notebooks/curation_api/python/src/dataset.py:26]: >Presumably, they know the difference and can save us from doing this work on the backend

We can _determine_ the difference and can therefore save _them_ from the hassle on the client side 😉 

But seriously, I think it's better this way, just like when they pass us a Collection uuid we don't want to make them also tell us whether it's a private or public Collection. It's unnecessary work for them, and it'll lead to accidental errors when they change an identifier and forget to change the type that they would need to pass in as well.

Comment (brianraymor) [reply notebooks/curation_api/python/src/dataset.py:26]: > We can determine the difference and can therefore save them from the hassle on the client side

+1. Curator time is valuable.

File: notebooks/curation_api/python/src/collection.py [hunk 6]
```diff
@@ -82,3 +82,18 @@ def get_collections(visibility: str = "PUBLIC") -> list:
...
     res.raise_for_status()
     return res.json().get("collections")
+
+
+def update_collection(collection_uuid: str, collection_form_metadata: dict) -> None:
+
+    url = url_builder(f"/collections/{collection_uuid}")
+    headers = get_headers()
+    try:
+        res = requests.put(url, headers=headers, data=json.dumps(collection_form_metadata))
```
Comment (nayib-jose-gloria) [inline notebooks/curation_api/python/src/collection.py:92]: probably already on your radar, but just flagging that this is a `patch` request now with Trent's latest merged PR!

**Review by Bento007 — CHANGES_REQUESTED** [review]

File: notebooks/curation_api/python/create_dataset_from_upload_link.ipynb [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+   "source": [
+    "set_api_access_config(api_key_file_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6085d322",
+   "metadata": {},
+   "source": [
+    "### Delete Dataset"
```
[the remark this answers is not in the document]
Comment (danieljhegeman) [reply notebooks/curation_api/python/create_dataset_from_upload_link.ipynb:131]: Ah, missed another change there. Good catch!

**Comment by danieljhegeman** [comment]
@Bento007 do you have any further comments on this or can I merge

**Review by Bento007 — APPROVED** [review]

**Pull request merged**